core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -162,12 +162,18 @@ public void start(final Listener<RespT> observer, Metadata headers) {
       // Context is already cancelled so no need to create a real stream, just notify the observer
       // of cancellation via callback on the executor
       stream = NoopClientStream.INSTANCE;
-      callExecutor.execute(new ContextRunnable(context) {
+      class ClosedByContext extends ContextRunnable {
+        ClosedByContext() {
```

Comment:
This is unfortunately verbose to improve the test failure output. But it probably is more useful overall, even if we could semi-easily have a better `toString()` for each `Runnable`. I guess it's just a cost we accept.